javascript-challenges.test.js

```diff
@@ -40,4 +100,41 @@ const hand4 = [7, 2, 7, 2, 7]
 
 // b) Create the function that makes the test pass.
 
+const fullHouse = (array) => {
+    if (array.length !== 5) {
+        return false;
+    }
+    const countMap = new Map();
+    for (let num of array) {
+        countMap.set(num, (countMap.get(num) || 0) + 1);
+    }
+    let pair = false;
+    let threeOfAKind = false;
+    for (let count of countMap.values()) {
+        if (count === 2) {
+            pair = true;
+        } else if (count === 3) {
+            threeOfAKind = true;
+        }
+    }
+    return pair && threeOfAKind;
+}
+
+console.log(fullHouse(hand1))
+console.log(fullHouse(hand2))
+console.log(fullHouse(hand3))
+console.log(fullHouse(hand4))
+
 // Pseudo code:
+// Input: Array of 5 numbers
+// Output: Full house or not a full hosue
+// Create a function that takes in an array
+// Ensure the array is exactly 5 numbers otherwise return false
+// Use countMap to count the occurrences of each number in the array
+// The for each num in array iterates through each number and updates the count
+// The second for loop checks to see if there is a count === 2 and a value ===3 if so set to true
+// Use the && to make sure the array meets both expressions
+
+
+// Somewhat grasped this one but the countMap isn't making 100% sense yet.
```

Comment:
Yeah this looks like a bit of over-engineering. I think you could refactor using the regular map method and make this code a little more understandable.